marklogic-client-api/src/main/java/com/marklogic/client/DatabaseClientFactory.java

```diff
@@ -1037,7 +1098,7 @@ private SSLContext createSSLContext()
         }
         keyManagerFactory.init(keyStore, certPassword.toCharArray());
         keyMgr = keyManagerFactory.getKeyManagers();
-        sslContext = SSLContext.getInstance("TLSv1.2");
+        sslContext = SSLContext.getInstance(sslProtocol);
```

Comment:
For safety reasons, I think this should say "if protocol is null, then use the default value".

Reply:
The null check needs to be there though. The DatabaseClientPropertySource is just one way of building a CertificateAuthContext. A user can build one themselves.